test/integration/lib/integration-server.ts

```diff
@@ -0,0 +1,182 @@
+import Koa from 'koa'
+import bodyParser from '@koa/bodyparser'
+import http from 'http'
+import { v4 as uuid } from 'uuid'
+import {
+  AccountProvider,
+  WebhookEventType,
+  Webhook
+} from 'mock-account-service-lib'
+import { TestConfig } from './config'
+import { AdminClient } from './admin-client'
+
+export class IntegrationServer {
+  private config: TestConfig
+  private app: Koa
+  private server?: http.Server
+  public webhookEventHandler: WebhookEventHandler
+
+  constructor(
+    config: TestConfig,
+    adminClient: AdminClient,
+    accounts: AccountProvider
+  ) {
+    this.config = config
+    this.app = new Koa()
+    this.app.use(bodyParser())
+    this.webhookEventHandler = new WebhookEventHandler(adminClient, accounts)
+  }
+
+  public start(port: number): void {
+    this.app.use(async (ctx) => {
+      if (ctx.path === '/webhooks' && ctx.method === 'POST') {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const { body } = ctx.request as Koa.Request & { body?: any }
+
+        if (this.isWebhookEvent(body)) {
+          this.webhookEventHandler.handleWebhookEvent(body)
+          ctx.status = 200
+          ctx.body = 'Webhook received'
+        } else {
+          ctx.status = 400
+          ctx.body = 'Invalid WebhookEvent payload'
+        }
+      }
+
+      if (ctx.path === '/rates' && ctx.method === 'GET') {
+        const { base } = ctx.query
+        ctx.body = {
+          base,
+          rates: {
+            ...this.config.seed.rates[base as string]
+          }
+        }
+      }
+    })
+
+    this.server = this.app.listen(port, () => {
+      console.log(`Integration server listening on port ${port}`)
```

Comment:
I think for the next PR we should just make this a pino logger and set it to silent during tests

Reply:
yeah we probably dont even need these tbh. will add it to the cleanup to address one way or another.